### PR TITLE
[Doc] refine evaluation docs

### PR DESCRIPTION
**Pull request by ly015**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refine the tutorial and design documents of the evaluator and metric.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/tutorials/evaluation.md [hunk 1]
```diff
@@ -37,11 +37,83 @@ test_evaluator = [
 
 ### 自定义评测指标
 
-如果算法库中提供的常用评测指标无法满足需求，用户也可以增加自定义的评测指标。具体的方法可以参考[评测指标和评测器设计](../design/evaluation.md)。
+如果算法库中提供的常用评测指标无法满足需求，用户也可以增加自定义的评测指标。我们以简化的分类正确率为例，介绍实现自定义评测指标的方法：
+
+1. 在定义新的评测指标类时，需要继承基类 [`BaseMetric`](/mmengine/evaluator/metric.py)（关于该基类的介绍，可以参考[设计文档](/docs/zh_cn/design/evaluation.md)）。此外，评测指标类需要用注册器 `METRICS` 进行注册（关于注册器的说明请参考 [Registry 文档](/docs/zh_cn/tutorials/registry.md)）。
+
+2. 实现 `process()` 方法。该方法有 2 个输入参数，分别是一个批次的测试数据样本 `data_batch` 和模型预测结果 `predictions`。我们从中分别取出样本类别标签和分类预测结果，并存放在 `self.results` 中。
```
Comment (RangiLyu) [inline docs/zh_cn/tutorials/evaluation.md:44]: 在重构数据流之后，BaseMetric 的 process 输入变掉了
```suggestion
2. 实现 `process()` 方法。该方法有 2 个输入参数，分别是一个批次的测试数据样本 `data_batch` 和模型预测结果 `data_samples`。我们从中分别取出样本类别标签和分类预测结果，并存放在 `self.results` 中。
```

File: docs/zh_cn/tutorials/evaluation.md [hunk 2]
```diff
@@ -37,11 +37,83 @@ test_evaluator = [
...
+
+    Default prefix: ACC
+
+    Metrics:
+        - accuracy (float): classification accuracy
+    """
+
+    default_prefix = 'ACC'  # 设置 default_prefix
+
+    def process(self, data_batch: Sequence[dict], predictions: Sequence[dict]):
```
Comment (RangiLyu) [inline docs/zh_cn/tutorials/evaluation.md:71]:
```suggestion
    def process(self, data_batch: Sequence[dict], data_samples: Sequence[dict]):
```

File: docs/zh_cn/tutorials/evaluation.md [hunk 3]
```diff
@@ -37,11 +37,83 @@ test_evaluator = [
...
+        Args:
+            data_batch (Sequence[Tuple[Any, dict]]): A batch of data
+                from the dataloader.
+            predictions (Sequence[dict]): A batch of outputs from
+                the model.
+        """
+
+        # 取出分类预测结果和类别标签
+        result = {
+            'pred': predictions['pred_label'],
```
Comment (RangiLyu) [inline docs/zh_cn/tutorials/evaluation.md:85]: 新版数据流，pred和gt都在data_sample里取

File: docs/zh_cn/design/evaluation.md [hunk 4]
```diff
@@ -1,111 +1,88 @@
...
 
-`compute_metrics()` 方法有 1 个输入参数 `results`，里面存放了所有批次测试数据经过 `process()` 方法处理后得到的结果。从中取出样本类别标签和分类预测结果，即可计算得到分类正确率 `acc`。最终，将计算得到的评测指标以字典的形式返回。
+`BaseMetric` 有以下 2 个重要的方法需要在子类中重写：
 
-此外，我们建议在子类中为类属性 `default_prefix` 赋值。如果在初始化参数（即 config 中）没有指定 `prefix`，则会自动使用 `default_prefix` 作为评测指标名的前缀。同时，应在 docstring 中说明该评测指标类的 `default_prefix` 值以及所有的返回指标名称。
+- **`process()`** 用于处理每个批次的测试数据和模型预测结果。处理结果应存放在 `self.results` 列表中，用于在处理完所有测试数据后计算评测指标。该方法具有以下 2 个参数：
 
-具体的实现如下：
+  - `data_batch`：一个批次的测试数据样本
+  - `predictions`：对应的模型预测结果
```
Comment (C1rN09) [inline docs/zh_cn/design/evaluation.md:34]:
```suggestion
  - `data_batch`：一个批次的测试数据样本，通常直接来自于数据加载器
  - `data_samples`：对应的模型预测结果
```
另外下面还有一些 predictions，都一并改成 data_samples 吧

**Review by C1rN09 — APPROVED** [review]

**Review by RangiLyu — APPROVED** [review]

File: docs/zh_cn/tutorials/evaluation.md [hunk 5]
```diff
@@ -37,11 +37,83 @@ test_evaluator = [
 
 ### 自定义评测指标
 
-如果算法库中提供的常用评测指标无法满足需求，用户也可以增加自定义的评测指标。具体的方法可以参考[评测指标和评测器设计](../design/evaluation.md)。
+如果算法库中提供的常用评测指标无法满足需求，用户也可以增加自定义的评测指标。我们以简化的分类正确率为例，介绍实现自定义评测指标的方法：
+
+1. 在定义新的评测指标类时，需要继承基类 [`BaseMetric`](/mmengine/evaluator/metric.py)（关于该基类的介绍，可以参考[设计文档](/docs/zh_cn/design/evaluation.md)）。此外，评测指标类需要用注册器 `METRICS` 进行注册（关于注册器的说明请参考 [Registry 文档](/docs/zh_cn/tutorials/registry.md)）。
```
Comment (zhouzaida) [inline docs/zh_cn/tutorials/evaluation.md:42]:
```suggestion
1. 在定义新的评测指标类时，需要继承基类 [`BaseMetric`](mmengine.evaluator.BaseMetric)（关于该基类的介绍，可以参考[设计文档](../design/evaluation.md)）。此外，评测指标类需要用注册器 `METRICS` 进行注册（关于注册器的说明请参考 [Registry 文档](./registry.md)）。
```

File: docs/zh_cn/tutorials/evaluation.md [hunk 6]
```diff
@@ -1,21 +1,21 @@
 # 模型精度评测（Evaluation）
 
-在模型验证和模型测试中，通常需要对模型精度做定量评测。在 MMEngine 中实现了评测指标（Metric）和评测器（Evaluator）模块来完成这一功能：
-
-- 评测指标： 用于根据测试数据和模型预测结果，完成模型特定精度指标的计算。在 OpenMMLab 各算法库中提供了对应任务的常用评测指标，如 [MMClassification](https://github.com/open-mmlab/mmclassification) 中提供了[分类正确率指标（Accuracy）](https://mmclassification.readthedocs.io/zh_CN/dev-1.x/generated/mmcls.evaluation.Accuracy.html) 用于计算分类模型的 Top-k 分类正确率。
-
-- 评测器： 是评测指标的上层模块，用于在数据输入评测指标前完成必要的格式转换，并提供分布式支持。在模型训练和测试中，评测器由[执行器（Runner）](runner.md)自动构建。用户亦可根据需求手动创建评测器，进行离线评测。
+在模型验证和模型测试中，通常需要对模型精度做定量评测。我们可以通过在配置文件中指定\*\*评测指标（Metric）\*\*来实现这一功能。
```
Comment (zhouzaida) [inline docs/zh_cn/tutorials/evaluation.md:3]:
```suggestion
在模型验证和模型测试中，通常需要对模型精度做定量评测。我们可以通过在配置文件中指定 **评测指标（Metric）**来实现这一功能。
```

File: docs/zh_cn/design/evaluation.md [hunk 7]
```diff
@@ -1,111 +1,88 @@
...
+## 增加自定义评测指标
 
-        # 计算分类正确率
-        acc = (preds == gts).sum() / preds.size
+在 OpenMMLab 的各个算法库中，已经实现了对应方向的常用评测指标。如 MMDetection 中提供了 COCO 评测指标，MMClassification 中提供了 Accuracy、F1Score 等评测指标等。
 
-        # 返回评测指标结果
-        return {'accuracy': acc}
-```
+用户也可以增加自定义的评测指标。具体方法可以参考[教程文档](/docs/zh_cn/tutorials/evaluation.md#自定义评测指标)中给出的示例。
```
Comment (zhouzaida) [inline docs/zh_cn/design/evaluation.md:88]:
```suggestion
用户也可以增加自定义的评测指标。具体方法可以参考[教程文档](../tutorials/evaluation.md#自定义评测指标)中给出的示例。
```

**Review by zhouzaida — APPROVED** [review]

File: docs/zh_cn/tutorials/evaluation.md [hunk 8]
```diff
@@ -1,21 +1,21 @@
 # 模型精度评测（Evaluation）
 
-在模型验证和模型测试中，通常需要对模型精度做定量评测。在 MMEngine 中实现了评测指标（Metric）和评测器（Evaluator）模块来完成这一功能：
-
-- 评测指标： 用于根据测试数据和模型预测结果，完成模型特定精度指标的计算。在 OpenMMLab 各算法库中提供了对应任务的常用评测指标，如 [MMClassification](https://github.com/open-mmlab/mmclassification) 中提供了[分类正确率指标（Accuracy）](https://mmclassification.readthedocs.io/zh_CN/dev-1.x/generated/mmcls.evaluation.Accuracy.html) 用于计算分类模型的 Top-k 分类正确率。
-
-- 评测器： 是评测指标的上层模块，用于在数据输入评测指标前完成必要的格式转换，并提供分布式支持。在模型训练和测试中，评测器由[执行器（Runner）](runner.md)自动构建。用户亦可根据需求手动创建评测器，进行离线评测。
+在模型验证和模型测试中，通常需要对模型精度做定量评测。我们可以通过在配置文件中指定 **评测指标（Metric）**来实现这一功能。
```
Comment (zhouzaida) [inline docs/zh_cn/tutorials/evaluation.md:3]:
```suggestion
在模型验证和模型测试中，通常需要对模型精度做定量评测。我们可以通过在配置文件中指定评测指标（Metric）来实现这一功能。
```

**Pull request merged**